### PR TITLE
Refactor PermissionTemplateAccessesController

### PR DESCRIPTION
**Pull request by jcoyne**
Removes duplicate code and unnecessary methods.

### CODE REVIEW

File: app/controllers/hyrax/admin/permission_template_accesses_controller.rb [hunk 1]
```diff
@@ -29,13 +29,8 @@ def valid_delete?
...
         end
 
-        # @return [String] the identifier for the AdminSet for the currently loaded resource
-        def source_id
-          @source_id ||= @permission_template_access.permission_template.source_id
-        end
-
         def update_access(manage_changed:)
-          Forms::PermissionTemplateForm.new(@permission_template_access.permission_template).update_access(manage_changed: manage_changed)
+          permisison_template_form.update_access(manage_changed: manage_changed)
```
Comment (mjgiarlo) [inline app/controllers/hyrax/admin/permission_template_accesses_controller.rb]: Typo.

File: app/controllers/hyrax/admin/permission_template_accesses_controller.rb [hunk 2]
```diff
@@ -62,13 +57,18 @@ def after_destroy_error
...
-                                       .remove_access!(@permission_template_access)
+          permisison_template_form.remove_access!(@permission_template_access)
+        end
+
+        def permisison_template_form
+          @permission_template_form ||= Forms::PermissionTemplateForm.new(permission_template)
+        end
+
+        def permission_template
+          @permission_template ||= @permission_template_access.permission_template
```
Comment (mjgiarlo) [inline app/controllers/hyrax/admin/permission_template_accesses_controller.rb]: Should this use the memoizing method instead of the ivar?

Comment (jcoyne) [reply app/controllers/hyrax/admin/permission_template_accesses_controller.rb]: I don't know what memorizing method you are referring to. Can you help me out?

Comment (mjgiarlo) [reply app/controllers/hyrax/admin/permission_template_accesses_controller.rb]: Uhhh, I don't know. I completely miffed on this. nvm!

**Pull request merged**